File: order-producer/src/main/java/com/sahakav/coffee_shop/service/OrderCommandService.java
package com.sahakav.coffee_shop.service;

import com.sahakav.coffee_shop.model.CoffeeType;
import io.vavr.control.Try;

import java.util.UUID;

public interface OrderCommandService {

    Try<UUID> placeOrder(CoffeeType coffeeType);

}

File: order-producer/src/main/scala/com/sahakav/coffee_shop/service/OrderCommandService.scala
package com.sahakav.coffee_shop.service

import java.util.UUID

import com.sahakav.coffee_shop.model.{CoffeeOrderState, CoffeeType}

import scala.util.Try

trait OrderCommandService {


  def placeOrder(coffeeType: CoffeeType): Try[UUID]

  def getOrderState(orderId:UUID):CoffeeOrderState

}
